src/notes.c

```diff
@@ -491,6 +497,50 @@ int git_note_create(
 	return error;
 }
 
+int git_note_commit_create(
+	git_commit **notes_commit_out,
+	git_oid *notes_blob_out,
+	git_repository *repo,
+	git_commit *notes_commit,
+	const git_signature *author,
+	const git_signature *committer,
+	const git_oid *oid,
+	const char *note,
+	int allow_note_overwrite)
+{
+	int error;
+	git_tree *tree = NULL;
+	char *target = NULL;
+	git_oid note_commit_oid, note_blob_oid;
+
+	target = git_oid_allocfmt(oid);
+	GITERR_CHECK_ALLOC(target);
+
+	if (notes_commit != NULL && (error = git_commit_tree(&tree, notes_commit)) < 0)
+		goto cleanup;
+
+	error = note_write(&note_commit_oid, &note_blob_oid, repo, author, committer, NULL,
+			note, tree, target, &notes_commit, allow_note_overwrite);
+
+	if (error < 0)
+		goto cleanup;
+
+	if (notes_commit_out != NULL) {
+		error = git_commit_lookup(notes_commit_out, repo, &note_commit_oid);
+		if (error < 0)
+			goto cleanup;
+	}
```

Comment:
So we actually do a separate lookup on this commit. I'd instead propose simply giving out the OID of the commit, as with the blob ID, and have the caller handle the lookup in case it is required.